src/core/lib/event_engine/tcp_socket_utils.cc

```diff
@@ -224,10 +278,15 @@ absl::optional<int> ResolvedAddressIsWildcard(
 absl::StatusOr<std::string> ResolvedAddressToNormalizedString(
     const EventEngine::ResolvedAddress& resolved_addr) {
   EventEngine::ResolvedAddress addr_normalized;
-  if (ResolvedAddressIsV4Mapped(resolved_addr, &addr_normalized)) {
-    return ResolvedAddressToString(addr_normalized);
+  if (!ResolvedAddressIsV4Mapped(resolved_addr, &addr_normalized)) {
+    addr_normalized = resolved_addr;
   }
-  return ResolvedAddressToString(resolved_addr);
+  auto scheme = GetScheme(addr_normalized);
+  GRPC_RETURN_IF_ERROR(scheme.status());
+  if (*scheme == "unix") {
+    return ResolvedAddrToUriUnixIfPossible(&addr_normalized, true);
```

Comment:
The `ResolvedAddressToURI` function relies on the family of `To*String` functions, and they serve slightly different purposes. For a `To*String` function to now rely on and return a URI adds some confusing inconsistency. I'd recommend having `ResolvedAddressToNormalizedString` handle Unix sockaddrs internally. And if a client needs a URI, they should call a function that explicitly returns a URI (not a `To*String` function).

Reply:
There was some logic which was re-usable. Created a ResolvedAddrToUnixPathIfPossible internal function now to be used by To*String functions.